pkg/apis/workflow/v1alpha1/common.go

```diff
@@ -4,31 +4,26 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// TemplateGetter is an interface to get templates.
-type TemplateGetter interface {
+type ResourceScope string
+
+const (
+	ResourceScopeLocal      ResourceScope = "local"
+	ResourceScopeNamespaced ResourceScope = "namespaced"
+	ResourceScopeCluster    ResourceScope = "cluster"
+)
+
+// TemplateHolder is an object that holds templates; e.g. Workflow, WorkflowTemplate, and ClusterWorkflowTemplate
+type TemplateHolder interface {
```

Comment:
Renamed to `TemplateHolder` since it is more accurate.

It would be correct to say that a `Workflow` and a `WorkflowTemplate` "hold" templates. It would not be correct to say they "get" templates.